aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -310,6 +319,16 @@ Tensor& mvlgamma_(Tensor& self, int64_t p) {
     return result;                                                     \
   }
 
+// IMPLEMENT_UNARY_FLOATING_OP_* macros are used for ops which support 
+// implicit type promotion (integral and boolean inputs to default floating type)
+// Check PR #33322 for more details on this
+#define IMPLEMENT_UNARY_FLOATING_OP_CORE(op)                           \
+  Tensor op(const Tensor& self) {                                      \
+    Tensor result = isIntegralType(self.scalar_type(), /*include_bool=*/ true) ? at::empty({0}, self.options().dtype(typeMetaToScalarType(c10::get_default_dtype()))) : at::empty({0}, self.options()); \
```

Comment:
Line too long.

Reply:
The recent commit (https://github.com/pytorch/pytorch/pull/33322/commits/c816f39f45369536c2ffb329756ae36405b21d64) should solve this! :)